must-gather/collection-scripts/gather_ceph_resources

```diff
@@ -205,14 +205,31 @@ for ns in $namespaces; do
 
     oc delete -f pod_helper.yaml
 
+    crash_collection(){
+        oc debug nodes/"${node}" --to-namespace="${ns}" -- bash -c "sleep 5m" &
+        while [ "$(oc get pods -n openshift-storage | grep "${node//./}-debug" | awk '{print $2}')" != "1/1" ] ; do
+            sleep 3
+            echo "waiting for the debug pod to be in ready state"
+        done
+        oc rsync -n "${ns}" "$(oc get pods -n "${ns}"| grep "${node//./}-debug"| awk '{print $1}')":/host/var/lib/rook/openshift-storage/crash/ "${CRASH_OUTPUT_DIR}"
+    }
+
+    # creating a counter variable for collecting PID in array
+    count=1
     # Collecting ceph crash dump
     for node in $(oc get nodes -l cluster.ocs.openshift.io/openshift-storage='' --no-headers | awk '/Ready/{print $1}'); do
         printf "collecting crash logs  from node %s \n"  "${node}" | tee -a  "${BASE_COLLECTION_PATH}"/gather-debug.log
         CRASH_OUTPUT_DIR=${CEPH_COLLECTION_PATH}/crash_${node}
         mkdir -p "${CRASH_OUTPUT_DIR}"
-        oc debug nodes/"${node}" --to-namespace="${ns}" -- bash -c "sleep 5m" &
-        sleep 60
-        oc rsync -n "${ns}" "$(oc get pods -n "${ns}"| awk '/debug/{print $1}')":/host/var/lib/rook/openshift-storage/crash/ "${CRASH_OUTPUT_DIR}"
-        wait
+        crash_collection &
+        # collecting PID for bg jobs
+        pids[${count}]=$!
```

Comment:
I can't find where this variable is defined outside the `for` loop. Is it missing or am I missing it?

Reply:
line no 218

Reply:
That's `count`, not `pids`.

Reply:
pids not declared separately. This works fine.

Reply:
...how is this possible? If you don't declare an array outside the scope of the for loop it should disappear once the for loop exits.

Reply:
Okay, so I wrote a test script:
```bash
  #!/bin/bash
  
  idx=0
  for i in 1 2 3; do
      echo $i &
      pids[${idx}]=$!
      idx=${idx}+1
  done
  
  # wait for all pids
  echo "waiting for ${pids[*]} to terminate"
  wait "${pids[@]}"
```
And found that indeed `pids` is preserved outside the scope of the for loop. This comes as a bit of a surprise, but okay.

I will point out one thing, though: I was able to get rid of the `expr` call and just do basic variable arithmetic. Is there any reason you couldn't also do that? Would that resolve the shellcheck errors?

Reply:
I was not able to do that without expr but if you are able then I might have missed something. let me check that again.

Reply:
certain things were made for my initial logic after the review code changes are done. So the dependent code changes can also be modify.